src/windows/volume.cc

```diff
@@ -65,6 +65,41 @@ HRESULT drivelist::volume::GetReadOnlyFlag(const wchar_t letter, BOOL *out) {
   return result;
 }
 
+HRESULT drivelist::volume::HasFileSystem(const wchar_t letter, BOOL *out) {
```

Comment:
After some research, looks like the only way to reliably detect this is to run `GetVolumeInformation` and check if the error is `ERROR_UNRECOGNIZED_VOLUME`